core/src/main/java/org/apache/brooklyn/core/mgmt/BrooklynTags.java

```diff
@@ -51,9 +53,42 @@ public String getContents() {
             return contents;
         }
     }
+
+    public static class InterfacesTag {
+        @JsonProperty
+        final List<Class<?>> interfaces;
```

Comment:
Have you looked at the persistence files, how are classes serialized? Could be leaking a lot of state. Does it rebind cleanly if classes are coming from osgi bundles?
Consider converting to strings.